app/v2/notifications/get_notifications.py

```diff
@@ -13,10 +24,45 @@ def get_notification_by_id(notification_id):
     notification = notifications_dao.get_notification_with_personalisation(
         authenticated_service.id, notification_id, key_type=None
     )
-
     return jsonify(notification.serialize()), 200
 
 
+@v2_notification_blueprint.route('/<notification_id>/pdf', methods=['GET'])
+def get_pdf_for_notification(notification_id):
+    _data = {"notification_id": notification_id}
+    validate(_data, notification_by_id)
+    notification = notifications_dao.get_notification_by_id(
+        notification_id, authenticated_service.id, _raise=True
+    )
+
+    if notification.notification_type != LETTER_TYPE:
+        raise BadRequestError(message="Notification is not a letter", status_code=400)
+
+    if notification.status in {
+        NOTIFICATION_PENDING_VIRUS_CHECK,
+        NOTIFICATION_VIRUS_SCAN_FAILED,
+        NOTIFICATION_TECHNICAL_FAILURE,
```

Comment:
Is there a situation where the letter is in tech-failure but can still retrieve the letter?

Reply:
i thought best to play it safe, i don't know where the letter is (it could be in any bucket), or why it tech failured (could be something wrong with the letter for example). It might not even have passed virus scan.